collect_app/src/main/java/org/odk/collect/android/database/entities/DatabaseEntitiesRepository.kt

```diff
@@ -284,6 +284,7 @@ class DatabaseEntitiesRepository(context: Context, dbPath: String) : EntitiesRep
                     SELECT *, i.$ROW_ID
                     FROM $list e, ${getRowIdTableName(list)} i
                     WHERE e._id = i._id
+                    ORDER BY i.$ROW_ID
```

Comment:
I don't think we strictly need the `ORDER BY` here and in the other `queryWithAttachedRowId` overload and I can't work out a way to test it (I think the query is structured so that it always happens to works out). I feel safer having it there though so.